### PR TITLE
Upgrade Vert.x 3.9.1; mention vertx-stack-depchain, jackson

### PR DESCRIPTION
**Pull request by julianladisch**
Upgrade Vert.x to the latest released version 3.9.1 that contains several
bug fixes:
* https://github.com/vert-x3/wiki/wiki/3.9.1-Release-Notes
* https://github.com/vert-x3/wiki/wiki/3.9.1-Deprecations-and-breaking-changes

Add to upgrading notes
* Upgrade to Vert.x 3.9.1
* Using vertx-stack-depchain
* Dropping unused jackson-databind dependency

### CODE REVIEW

File: doc/upgrading.md [hunk 1]
```diff
@@ -58,6 +58,10 @@ See the [NEWS](../NEWS.md) summary of changes for each version.
   JSON files, otherwise GenerateRunner/SchemaDereferencer will fail with
   InvocationTargetException/DecodeException "Failed to decode".
   Hint: Use `for i in *; do jq empty $i || echo $i; done` to list invalid JSONs.
+* Update Vert.x to 3.9.1, consider using
+  [Bill of Materials (Maven BOM)](https://github.com/vert-x3/vertx-stack#bills-of-materials-maven-bom)
+  for io.vertx, com.fasterxml.jackson.core and com.fasterxml.jackson.dataformat dependencies. Remove jackson-databind
+  dependency from pom.xml if module doesn't use it directly.
```
Comment (adamdickmeiss) [inline doc/upgrading.md:65]: Should be a 30.1 change or 31. Hardly a bug fix, is it?

Comment (julianladisch) [reply doc/upgrading.md:65]: I don't agree.
Upgrading Vert.x from 3.9.0 to 3.9.1 is a bugfix.
Upgrading jackson-core from 2.10.1 to 2.10.2 is a bugfix.
Upgrading jackson-databind from 2.10.1 to 2.10.2. is a bugfix.
Upgrading jackson-annotations from 2.10.1 to 2.10.2 is a bugfix.
Upgrading jackson-dataformat-yaml from 2.10.1 to 2.10.2 is a bugfix.
Can you be more specific which upgrade you don't consider to be a bugfix?

Comment (adamdickmeiss) [reply doc/upgrading.md:65]: Unfortunately, Vert.x does not follow semantic versioning. https://github.com/vert-x3/wiki/wiki/3.9.1-Deprecations-and-breaking-changes

Comment (adamdickmeiss) [reply doc/upgrading.md:65]: OK.. I *think* the changed behavior does not affect us.. If that's the case, I'm happy we cherry-pick for the 30.0.1 release.

Comment (julianladisch) [reply doc/upgrading.md:65]: Caching only affects performance, and it does not affect us.

Vert.x uses semantic versioning in this way:

> Please note that the public API can only be broken on major releases. For any minor or point releases any changes to the API must be API compatible.

https://github.com/eclipse-vertx/vert.x/wiki/Guide-for-module-maintainers#api-compatibility

Comment (adamdickmeiss) [reply doc/upgrading.md:65]: Good. I have approved already

**Pull request merged**